adr/0023-required-optional.md

```diff
@@ -0,0 +1,31 @@
+# 23. Required -> Optional
+
+Date: 2024-01-02
+
+## Status
+
+Pending
+
+## Context
+
+> Feature request: <https://github.com/defenseunicorns/zarf/issues/2059>
+
+Currently, all Zarf components default to being optional due to the `required` key being _optional_ in the yaml. This leads to behaviors where a package author must ensure they are careful to annotate this key for each component--though the validation doesn't require them to so assumes a sort of "all things are optional" default state.
```

Comment:
```suggestion
Currently, all Zarf components default to being optional due to the `required` key being _optional_ in the yaml. This leads to package authors needing to ensure that they annotate this key for each component, and since nothing in the current validations prompts them about this they may be confused about the "all things are optional" default state.
```

Reply:
could be more concise